src/hooks/api/getters/useFeatures/useFeatures.ts

```diff
@@ -1,40 +1,39 @@
 import useSWR, { mutate, SWRConfiguration } from 'swr';
-import { useState, useEffect } from 'react';
+import { useCallback } from 'react';
 import { formatApiPath } from '../../../../utils/format-path';
 import handleErrorResponses from '../httpErrorResponseHandler';
+import { IFeatureToggle } from '../../../../interfaces/featureToggle';
 
-const useFeatures = (options: SWRConfiguration = {}) => {
-    const fetcher = async () => {
-        const path = formatApiPath('api/admin/features/');
-        return fetch(path, {
-            method: 'GET',
-        })
-            .then(handleErrorResponses('Features'))
-            .then(res => res.json());
-    };
-
-    const FEATURES_CACHE_KEY = 'api/admin/features/';
-
-    const { data, error } = useSWR(FEATURES_CACHE_KEY, fetcher, {
-        ...options,
-    });
+const PATH = formatApiPath('api/admin/features');
 
-    const [loading, setLoading] = useState(!error && !data);
+export interface UseFeaturesOutput {
+    features: IFeatureToggle[];
+    refetchFeatures: () => void;
+    loading: boolean;
+    error?: Error;
+}
 
-    const refetchFeatures = () => {
-        mutate(FEATURES_CACHE_KEY);
-    };
+export const useFeatures = (options?: SWRConfiguration): UseFeaturesOutput => {
+    const { data, error } = useSWR<{ features: IFeatureToggle[] }>(
+        PATH,
+        fetchFeatures,
+        options
+    );
 
-    useEffect(() => {
-        setLoading(!error && !data);
-    }, [data, error]);
+    const refetchFeatures = useCallback(() => {
```

Comment:
There's an interesting conversation to be had here!

Is this actually faster than not using useCallback? I'm asking because I don't actually know. I read an article on this topic a year back that made me think about these kinds of optimizations. I think it was [this one](https://kentcdodds.com/blog/usememo-and-usecallback)

Now, personally I'm really into performance optimizations. I find that it is especially important with the amount of JS we are putting in the browser these days, but using these tools do add complexity so I think we should think about when and where to use them. React is so fast these days, and the v8 engine caches functions where inputs do not change regardless of what we do in react. 

If it makes sense, then we should of course add it. But we should also keep this in mind: 
(1) Not doing stuff is faster than doing stuff 
(2) You don't have a performance problem until you have a performance problem

Reply:
Good point! I didn't really consider the local performance aspect. My only goal here was for the hook to return a stable reference to `refetchFeatures`, so that it plays well with things like `React.memo`. Always nice when hooks return stable stuff, easier for consumers.

Reply:
Ahh, that makes sense.